app/gui/controller/double-representation/src/context_switch.rs

```diff
@@ -0,0 +1,214 @@
+//! A module containing utilities for managing execution context switch expressions.
+//!
+//! The context switch expressions are used to enable or disable a certain context for a single
+//! node.
+//!
+//! For example, the following code:
+//! ```ignore
+//! operator11 = Runtime.with_enabled_context Context.Output environment="design" <|
+//!     operator10.write "C:\Temp\Test.xlsx"
+//! ```
+//! will enable the [`Context::Output`] context for the `operator11` node in `design` environment.
+
+use crate::prelude::*;
+
+use crate::name::QualifiedName;
+use crate::name::QualifiedNameRef;
+
+use ast::known;
+use ast::opr::qualified_name_chain;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/// FQN for the [`Context::Output`].
+const OUTPUT: [&str; 5] = ["Standard", "Base", "Runtime", "Context", "Output"];
+/// FQN for the [`ContextSwitch::Enable`].
+const ENABLE: [&str; 4] = ["Standard", "Base", "Runtime", "with_enabled_context"];
+/// FQN for the [`ContextSwitch::Disable`].
+const DISABLE: [&str; 4] = ["Standard", "Base", "Runtime", "with_disabled_context"];
+
+// ===============================
+// === ContextSwitchExpression ===
+// ===============================
+
+// === ContextSwitch
+
+/// The two possible context switches.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[allow(missing_docs)]
+pub enum ContextSwitch {
+    Enable,
+    Disable,
+}
+
+impl<'a> TryFrom<QualifiedNameRef<'a>> for ContextSwitch {
+    type Error = ();
```

Comment:
I would use a normal error message. You can use anyhow::Error (we already started using it somewhere else).

Reply:
Done